fabfile.py

```diff
@@ -1,197 +1,518 @@
-    returns opal-referral
+
+def postgres_command(command, capture=False):
+    return local(
+        '{0} "{1}"'.format(DB_COMMAND_PREFIX, command),
+        capture=capture
+    )
+
+
+def postgres_create_database(some_env):
+    """ creates a database and user if they don't already exist.
+        the db_name is created from the release name
     """
 
-    if "github" in some_url and "opal" in some_url:
-        package_name = some_url.split("@")[0].split("/")[-1]
-        package_name = package_name.replace(".git", "")
-        version = some_url.split("@")[-1].split("#")[0]
-        return {package_name: version}
-
-
-def checkout(package_name_version):
-    with lcd(fabfile_dir):
-        with lcd(".."):
-            existing_packages = local("ls", capture=True).split("\n")
-            uncommitted = []
-
-            for package_name, version in package_name_version.iteritems():
-                if package_name in existing_packages:
-                    with lcd(package_name):
-                        if check_for_uncommitted():
-                            uncommitted.append(package_name)
-
-            if len(uncommitted):
-                print "we have uncommited changes in {} quitting".format(
-                    ", ".join(uncommitted)
-                )
-                return
-
-            for package_name, version in package_name_version.iteritems():
-                if package_name not in existing_packages:
-                    print "cloning {}".format(package_name)
-                    local("git clone {}".format(package_name))
-                    with lcd(package_name):
-                        print "checking out {0} to {1}".format(
-                            package_name, version
-                        )
-                else:
-                    local("git fetch origin")
-
-                with lcd(package_name):
-                    local("git checkout {}".format(version))
-                    local("git pull origin {}".format(version))
-                    local("python setup.py develop")
-
-
-def db_commands(username):
-    cmds =  [
-        "python manage.py migrate",
-        "python manage.py loaddata data/elcid.teams.json",
-        "python manage.py load_lookup_lists -f data/lookuplists/lookuplists.json",
-        "python manage.py create_random_data"
-    ]
-    python_cmd = "echo '"
-    python_cmd += "from django.contrib.auth.models import User; "
-    python_cmd += "User.objects.create_superuser(\"{0}\", \"{0}@example.com\", \"{0}1\")".format(username)
-    python_cmd += "' | python ./manage.py shell"
-    cmds.append(python_cmd)
-    return cmds
-
-def push_to_heroku(remote_name):
-    with lcd(fabfile_dir):
-        current_branch_name = local(
-            "git rev-parse --abbrev-ref HEAD", capture=True
+    #  https://stackoverflow.com/questions/14549270/check-if-database-exists-in-postgresql-using-shell
+    command = "sudo -u postgres psql -lqt | cut -d \| -f 1 | grep -w {database_name} | wc -l"
```

Comment:
This feels a bit `from stackoverflow import database_exists`...
(https://github.com/drathier/stack-overflow-import) I'm not sure I understand why this works..